cmd/config/etcd/etcd.go

```diff
@@ -36,61 +36,100 @@ const (
 
 // etcd environment values
 const (
+	Endpoints     = "endpoints"
+	CoreDNSPath   = "coredns_path"
+	ClientCert    = "client_cert"
+	ClientCertKey = "client_cert_key"
+
+	EnvEtcdState         = "MINIO_ETCD_STATE"
 	EnvEtcdEndpoints     = "MINIO_ETCD_ENDPOINTS"
+	EnvEtcdCoreDNSPath   = "MINIO_ETCD_COREDNS_PATH"
 	EnvEtcdClientCert    = "MINIO_ETCD_CLIENT_CERT"
 	EnvEtcdClientCertKey = "MINIO_ETCD_CLIENT_CERT_KEY"
 )
 
-// New - Initialize new etcd client
-func New(rootCAs *x509.CertPool) (*clientv3.Client, error) {
-	envEndpoints := env.Get(EnvEtcdEndpoints, "")
-	if envEndpoints == "" {
-		// etcd is not configured, nothing to do.
+// DefaultKVS - default KV settings for etcd.
+var (
+	DefaultKVS = config.KVS{
+		config.State:   config.StateOff,
+		config.Comment: "This is a default etcd configuration",
+		Endpoints:      "",
+		CoreDNSPath:    "/skydns",
+		ClientCert:     "",
+		ClientCertKey:  "",
+	}
+)
+
+// Config - server etcd config.
+type Config struct {
+	Enabled     bool   `json:"enabled"`
+	CoreDNSPath string `json:"coreDNSPath"`
+	clientv3.Config
+}
+
+// New - initialize new etcd client.
+func New(cfg Config) (*clientv3.Client, error) {
+	if !cfg.Enabled {
 		return nil, nil
 	}
+	return clientv3.New(cfg.Config)
+}
+
+// LookupConfig - Initialize new etcd config.
+func LookupConfig(kv config.KVS, rootCAs *x509.CertPool) (Config, error) {
+	cfg := Config{}
+	if err := config.CheckValidKeys(config.EtcdSubSys, kv, DefaultKVS); err != nil {
+		return cfg, err
+	}
 
-	etcdEndpoints := strings.Split(envEndpoints, config.ValueSeparator)
+	stateBool, err := config.ParseBool(env.Get(EnvEtcdState, kv.Get(config.State)))
+	if err != nil {
+		return cfg, err
+	}
+
+	endpoints := env.Get(EnvEtcdEndpoints, kv.Get(Endpoints))
+	if stateBool && len(endpoints) == 0 {
+		return cfg, config.Error("'endpoints' key cannot be empty if you wish to enable etcd")
+	}
+
+	if len(endpoints) == 0 {
+		return cfg, nil
+	}
+
+	cfg.Enabled = true
```

Comment:
this can be set after checking that the endpoints is not set to something like ",,,,"

Reply:
Resolving this as we agreed to move forward and address these things later.